Build failure: yabridge

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

- Build `yabridge` with `nixpkgs` commit [ec7c70d12ce2](https://github.com/NixOS/nixpkgs/commit/ec7c70d12ce2fc37cb92aff673dcdca89d187bae) : `nix-build --no-out-link '<nixpkgs>' -A yabridge`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/321715352

### Relevant log output

```console
[436/436] Linking target yabridge-host-32
FAILED: [code=2] yabridge-host-32 
wineg++  -o yabridge-host-32 yabridge-host-32.p/src_common_communication_vst2.cpp.o yabridge-host-32.p/src_common_serialization_vst2.cpp.o yabridge-host-32.p/src_common_configuration.cpp.o yabridge-host-32.p/src_common_logging_common.cpp.o yabridge-host-32.p/src_common_logging_vst2.cpp.o yabridge-host-32.p/src_common_audio-shm.cpp.o yabridge-host-32.p/src_common_plugins.cpp.o yabridge-host-32.p/src_common_process.cpp.o yabridge-host-32.p/src_common_utils.cpp.o yabridge-host-32.p/src_include_llvm_small-vector.cpp.o yabridge-host-32.p/src_wine-host_bridges_common.cpp.o yabridge-host-32.p/src_wine-host_bridges_group.cpp.o yabridge-host-32.p/src_wine-host_bridges_vst2.cpp.o yabridge-host-32.p/src_wine-host_editor.cpp.o yabridge-host-32.p/src_wine-host_host.cpp.o yabridge-host-32.p/src_wine-host_utils.cpp.o yabridge-host-32.p/src_wine-host_xdnd-proxy.cpp.o yabridge-host-32.p/src_common_logging_clap.cpp.o yabridge-host-32.p/src_common_serialization_clap_ext_audio-ports.cpp.o yabridge-host-32.p/src_common_serialization_clap_ext_audio-ports-config.cpp.o yabridge-host-32.p/src_common_serialization_clap_ext_note-name.cpp.o yabridge-host-32.p/src_common_serialization_clap_ext_note-ports.cpp.o yabridge-host-32.p/src_common_serialization_clap_ext_params.cpp.o yabridge-host-32.p/src_common_serialization_clap_events.cpp.o yabridge-host-32.p/src_common_serialization_clap_host.cpp.o yabridge-host-32.p/src_common_serialization_clap_plugin.cpp.o yabridge-host-32.p/src_common_serialization_clap_process.cpp.o yabridge-host-32.p/src_common_serialization_clap_stream.cpp.o yabridge-host-32.p/src_wine-host_bridges_clap-impls_host-proxy.cpp.o yabridge-host-32.p/src_wine-host_bridges_clap.cpp.o yabridge-host-32.p/src_common_logging_vst3.cpp.o yabridge-host-32.p/src_common_serialization_vst3_component-handler_component-handler.cpp.o yabridge-host-32.p/src_common_serialization_vst3_component-handler_component-handler-2.cpp.o yabridge-host-32.p/src_common_serialization_vst3_component-handler_component-handler-3.cpp.o yabridge-host-32.p/src_common_serialization_vst3_component-handler_component-handler-bus-activation.cpp.o yabridge-host-32.p/src_common_serialization_vst3_component-handler_progress.cpp.o yabridge-host-32.p/src_common_serialization_vst3_component-handler_unit-handler.cpp.o yabridge-host-32.p/src_common_serialization_vst3_component-handler_unit-handler-2.cpp.o yabridge-host-32.p/src_common_serialization_vst3_context-menu_context-menu.cpp.o yabridge-host-32.p/src_common_serialization_vst3_host-context_host-application.cpp.o yabridge-host-32.p/src_common_serialization_vst3_host-context_plug-interface-support.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plug-view_parameter-finder.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plug-view_plug-view.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plug-view_plug-view-content-scale-support.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plug-frame_plug-frame.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_audio-presentation-latency.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_audio-processor.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_automation-state.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_component.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_connection-point.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_edit-controller.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_edit-controller-2.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_edit-controller-host-editing.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_info-listener.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_keyswitch-controller.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_midi-learn.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_midi-mapping.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_note-expression-controller.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_note-expression-physical-ui-mapping.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_parameter-function-name.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_plugin-base.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_prefetchable-support.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_process-context-requirements.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_program-list-data.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_unit-data.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_unit-info.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin_xml-representation-controller.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin-factory_plugin-factory.cpp.o yabridge-host-32.p/src_common_serialization_vst3_attribute-list.cpp.o yabridge-host-32.p/src_common_serialization_vst3_base.cpp.o yabridge-host-32.p/src_common_serialization_vst3_bstream.cpp.o yabridge-host-32.p/src_common_serialization_vst3_component-handler-proxy.cpp.o yabridge-host-32.p/src_common_serialization_vst3_connection-point-proxy.cpp.o yabridge-host-32.p/src_common_serialization_vst3_context-menu-proxy.cpp.o yabridge-host-32.p/src_common_serialization_vst3_context-menu-target.cpp.o yabridge-host-32.p/src_common_serialization_vst3_event-list.cpp.o yabridge-host-32.p/src_common_serialization_vst3_host-context-proxy.cpp.o yabridge-host-32.p/src_common_serialization_vst3_message.cpp.o yabridge-host-32.p/src_common_serialization_vst3_param-value-queue.cpp.o yabridge-host-32.p/src_common_serialization_vst3_parameter-changes.cpp.o yabridge-host-32.p/src_common_serialization_vst3_physical-ui-map-list.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plug-frame-proxy.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plug-view-proxy.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin-proxy.cpp.o yabridge-host-32.p/src_common_serialization_vst3_plugin-factory-proxy.cpp.o yabridge-host-32.p/src_common_serialization_vst3_process-data.cpp.o yabridge-host-32.p/src_wine-host_bridges_vst3-impls_component-handler-proxy.cpp.o yabridge-host-32.p/src_wine-host_bridges_vst3-impls_connection-point-proxy.cpp.o yabridge-host-32.p/src_wine-host_bridges_vst3-impls_context-menu-proxy.cpp.o yabridge-host-32.p/src_wine-host_bridges_vst3-impls_host-context-proxy.cpp.o yabridge-host-32.p/src_wine-host_bridges_vst3-impls_plug-frame-proxy.cpp.o yabridge-host-32.p/src_wine-host_bridges_vst3.cpp.o -Wl,--as-needed -Wl,--no-undefined -mwindows -Wl,-rpath,/nix/store/mlissa4bdrvfa96nf38c7dp14aqhg07m-libxcb-1.17.0/lib -Wl,--start-group src/common/vst3/libvst3_sdk_hosting_wine_32bit.a src/common/vst3/libvst3_base_wine_32bit.a src/common/vst3/libvst3_pluginterfaces_wine_32bit.a -m32 -lrt -lole32 -lpthread /nix/store/mlissa4bdrvfa96nf38c7dp14aqhg07m-libxcb-1.17.0/lib/libxcb.so -lshell32 -luuid -Wl,--end-group
/nix/store/47mn80zqpygykqailwzw8zlag4cgl75q-binutils-2.44/bin/ld: skipping incompatible /nix/store/fk5m2lbwjkld3cgk53wzmdp8crcgzi7c-glibc-multi-2.42-47/lib/librt.so when searching for -lrt
/nix/store/47mn80zqpygykqailwzw8zlag4cgl75q-binutils-2.44/bin/ld: cannot find -lole32: No such file or directory
/nix/store/47mn80zqpygykqailwzw8zlag4cgl75q-binutils-2.44/bin/ld: skipping incompatible /nix/store/fk5m2lbwjkld3cgk53wzmdp8crcgzi7c-glibc-multi-2.42-47/lib/libpthread.so when searching for -lpthread
/nix/store/47mn80zqpygykqailwzw8zlag4cgl75q-binutils-2.44/bin/ld: cannot find -lshell32: No such file or directory
/nix/store/47mn80zqpygykqailwzw8zlag4cgl75q-binutils-2.44/bin/ld: cannot find -luuid: No such file or directory
collect2: error: ld returned 1 exit status
winegcc: /nix/store/s6kvx73nh6v9gyz0lkfxijw1lnwcz8qg-gcc-wrapper-15.2.0/bin/g++ failed
ninja: build stopped: subcommand failed.
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.7-zen1, NixOS, 26.05 (Yarara), 26.05.20260208.d6c7193`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.3`
 - nixpkgs: `/nix/store/gnf2ayqhb31qvgh1s9mg3aqaj9378cg7-source`



### Notify maintainers

@kira-bruneau 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ya/yabridge/hardcode-dependencies.patch
+++ b/pkgs/by-name/ya/yabridge/hardcode-dependencies.patch
@@ -1,16 +1,3 @@
-diff --git a/meson.build b/meson.build
-index 9e69128d..8c53ac88 100644
---- a/meson.build
-+++ b/meson.build
-@@ -226,7 +226,7 @@ if is_64bit_system
-   xcb_64bit_dep = dependency('xcb')
- endif
- if with_bitbridge
--  xcb_32bit_dep = winegcc.find_library('xcb')
-+  xcb_32bit_dep = winegcc.find_library('xcb', dirs: ['@libxcb32@/lib'])
- endif
- 
- # These are all headers-only libraries, and thus won't require separate 32-bit
 diff --git a/src/common/notifications.cpp b/src/common/notifications.cpp
 index 654b6c83..78ba2fe7 100644
 --- a/src/common/notifications.cpp

--- a/pkgs/by-name/ya/yabridge/package.nix
+++ b/pkgs/by-name/ya/yabridge/package.nix
@@ -1,9 +1,8 @@
 {
   lib,
-  multiStdenv,
+  stdenv,
   fetchFromGitHub,
   replaceVars,
-  pkgsi686Linux,
   dbus,
   meson,
   ninja,
@@ -73,7 +72,7 @@ let
     hash = "sha256-LsPHPoAL21XOKmF1Wl/tvLJGzjaCLjaDAcUtDvXdXSU=";
   };
 in
-multiStdenv.mkDerivation (finalAttrs: {
+stdenv.mkDerivation (finalAttrs: {
   pname = "yabridge";
   version = "5.1.1";
 
@@ -105,7 +104,6 @@ multiStdenv.mkDerivation (finalAttrs: {
     # Hard code bitbridge & runtime dependencies
     (replaceVars ./hardcode-dependencies.patch {
       libdbus = dbus.lib;
-      libxcb32 = pkgsi686Linux.libxcb;
       inherit wine;
     })
 
@@ -140,7 +138,7 @@ multiStdenv.mkDerivation (finalAttrs: {
   mesonFlags = [
     "--cross-file"
     "cross-wine.conf"
-    "-Dbitbridge=true"
+    "-Dbitbridge=false"
 
     # Requires CMake and is unnecessary
     "-Dtomlplusplus:generate_cmake_config=false"
@@ -149,7 +147,7 @@ multiStdenv.mkDerivation (finalAttrs: {
   installPhase = ''
     runHook preInstall
     mkdir -p "$out/bin" "$out/lib"
-    cp yabridge-host{,-32}.exe{,.so} "$out/bin"
+    cp yabridge-host.exe{,.so} "$out/bin"
     cp libyabridge{,-chainloader}-{vst2,vst3,clap}.so "$out/lib"
     runHook postInstall
   '';

